csrc/python_frontend/translation.cpp

```diff
@@ -635,6 +709,55 @@ class FusionTranslator : public OptInConstDispatch {
     map_val_to_fd_index_.emplace(bcast_op->out(), output());
   }
 
+  // Map SqueezeOp to python frontend
+  void handle(const SqueezeOp* sop) final {
+    std::vector<int64_t> squeeze_dims;
+    const std::vector<bool>& is_squeeze_dims = sop->getSqueezeDimFlags();
+    for (int64_t dim : c10::irange((int64_t)is_squeeze_dims.size())) {
+      if (is_squeeze_dims.at(dim)) {
+        squeeze_dims.push_back(dim);
+      }
+    }
+
+    // Always squeeze_expanded dimensions
+    Tensor output = fd_->defineTensor(sop->out()->as<TensorView>()->nDims());
+    map_val_to_fd_index_.emplace(sop->out(), output());
+    fd_->defineRecord(new SqueezeOpRecord(
+        {fd_->recordingState(map_val_to_fd_index_.at(sop->in()))},
+        {fd_->recordingState(output())},
+        squeeze_dims,
+        /*squeeze_expanded=*/true));
```

Comment:
This feels a bit strange here.

SqueezeOp doesn't have a field for `squeeze_expanded`. It reads like we added this field here just to avoid an assert during the translation.

Is there a reason that we wanted to disable squeezing expanded dimensions in the first place?

Reply:
IIUC, for reduction operations, we want to squeeze expanded dimensions to avoid actually computation.

Reply:
Alternative: Check if the squeeze occurs on any expanded dimensions. Then map to a reduction operation. You wouldn't know which reduction operation `(sum, prod, min, max)` was originally specified.

Reply:
> for reduction operations, we want to squeeze expanded dimensions to avoid actually computation.

I see what you mean now. So we still want to disable squeeze on expanded dimensions in general, since we want to raise exception when that occurs, similar to how we treat squeeze on non-broadcast dimensions.

I'm fine with the approach here as-is. Since this is only used as a check in the operation. Let's put in a comment here mentioning why it's ok to have this field default-on